contracts/v2/lib/DepositContractBase.sol

```diff
@@ -131,4 +136,38 @@ contract DepositContractBase {
 
         return node;
     }
+
+    /**
+     * @notice Validates that a frontier represents a valid subtree
+     * @dev Checks that frontier elements match Merkle proof siblings at appropriate heights
+     * @param subTreeLeafCount The number of leaves in the subtree
+     * @param subTreeFrontier The proposed frontier of the subtree
+     * @param currentTreeProof The Merkle proof siblings from the current tree
+     * @return true if the subtree frontier is valid
+     */
+    function _isValidSubtreeFrontier(
+        uint256 subTreeLeafCount,
+        bytes32[_DEPOSIT_CONTRACT_TREE_DEPTH] calldata subTreeFrontier,
+        bytes32[_DEPOSIT_CONTRACT_TREE_DEPTH] calldata currentTreeProof
+    ) internal pure returns (bool) {
+        // Verify subtree frontier consistency with the proof
+        uint256 index = subTreeLeafCount;
+        uint256 height = 0;
+
+        // Check each height where subtree frontier should have elements
+        while (index != 0 && height < _DEPOSIT_CONTRACT_TREE_DEPTH) {
+            if ((index & 1) == 1) {
+                // At this height, subtree has an element that must match proof sibling
+                if (subTreeFrontier[height] != currentTreeProof[height]) {
+                    return false; // Frontier element doesn't match proof
+                }
+            }
+            // If bit is 0, subtree doesn't have element at this height (skip check)
+
+            height++;
+            index >>= 1;
```

Comment:
guess this is the first approach, it should be marked as darft the PR

steps will be like::
Take the frist bit ( in reversed order) is different from both deposit counts.
Once that happen, proof inside that "branch" provide the merkle proof using the leftover frontier to get to the previous branch roort